dcs/liveries_scanner.py

```diff
@@ -235,9 +235,21 @@ def scan_lua_description(path: str, unit: str) -> None:
         """
         description_path = os.path.join(path, "description.lua")
         if os.path.exists(description_path):
-            with open(description_path, "r", encoding="utf-8") as file:
-                code = file.read()
-                Liveries.scan_lua_code(code, path, unit)
+            # Known encodings used for description.lua files
+            encodes = ["utf-8", "ansi"]
+            scanned = False
+            for enc in encodes:
+                with open(description_path, "r", encoding=enc) as file:
+                    try:
+                        code = file.read()
+                    except UnicodeDecodeError:
+                        file.close()
```

Comment:
This is unnecessary, btw. The `with` does this.

Reply:
> This is unnecessary, btw. The `with` does this.

I wasn't sure due to the continue statement, so I played it safe.

Should I remove it again in my other PR?

Reply:
It's NBD, just a tip for the future. https://docs.python.org/3/reference/datamodel.html#context-managers explains how `with` works if you're interested.

Reply:
Unfortunately the link you mentioned doesn't really answer my question in this specific case, however I can find numerous examples on stack-overflow where jump-statements are used from within a context manager. So theoretically I conclude that closing the file is redundant indeed, though this particular case gave me the creeps 😂

Anyway, after work I'll probably end up adjusting it 👍